crypto/fipsmodule/sha/sha3.c

```diff
@@ -312,18 +312,18 @@ int SHAKE_Final(uint8_t *md, KECCAK1600_CTX *ctx, size_t len) {
   ctx->state = KECCAK1600_STATE_FINAL;
 
   FIPS_service_indicator_update_state();
-
   return 1;
 }
 
 // SHAKE_Squeeze can be called multiple time for incremental XOF output
 int SHAKE_Squeeze(uint8_t *md, KECCAK1600_CTX *ctx, size_t len) {
+  size_t block_bytes;
+  ctx->md_size = len;
```

Comment:
Why did you move this? This is before the `ctx ==NULL` check now, can you add some more tests with invalid input to make sure we catch all of these issues.